lark/parsers/earley_forest.py

```diff
@@ -280,10 +280,10 @@ def visit(self, root):
         # of a symbol/intermediate so that we can process both up and down. Also,
         # since the SPPF can have cycles it allows us to detect if we're trying
         # to recurse into a node that's already on the stack (infinite recursion).
-        visiting = set()
+        visiting = OrderedSet()
```

Comment:
`visiting` and `visited` probably don't need to be ordered because they are never iterated over.